README.md

```diff
@@ -30,25 +51,230 @@ npm install --save ipfsd-ctl
+  const ipfsCtrl = ipfsd.ctrl
+  ipfsCtl.id(function (err, id) {
     console.log(id)
-    process.exit()
+    ipfsCtrl.stopDaemon()
   })
 })
 ```
 
-If you need want to use an existing ipfs installation you can set `$IPFS_EXEC=/path/to/ipfs` to ensure it uses that.
+### Remote node
+
+```js
+// Start a remote disposable node, and get access to the api
+// print the node id, and stop the temporary daemon
+
+const controllerFactory = require('ipfsd-ctl')
+const daemonFactory = controllerFactory()
+
+const port = 9999
+daemonFactory.start(port, (err) => {
+  if (err) {
+    throw err
+  }
+  
+  daemonFactory.spawn(function (err, ipfsd) {
+    const ipfsCtl = ipfsd.ctl
+    const ipfsCtrl = ipfsd.ctrl
+    ipfsCtl.id(function (err, id) {
+      console.log(id)
+      ipfsCtrl.stopDaemon()
+      server.stop()
+    })
+  })  
+})
+```
+
+It's also possible to start the server from `.aegir` `pre` and `post` hooks. 
+
+```js
+'use strict'
+
+const server = require('./src').server
+
+module.exports = {
+  karma: {
+    files: [{
+      pattern: 'test/fixtures/**/*',
+      watched: false,
+      served: true,
+      included: false
+    }],
+    singleRun: true
+  },
+  hooks: {
+    browser: {
+      pre: server.start,
+      post: server.stop
+    }
+  }
+}
+```
+
+## API
+
+### Daemon Factory
+
+#### Create factory
+
+```js
+const controllerFactory = require('ipfsd-ctl')
+const daemonFactory = controllerFactory()
+```
+
+> Create a factory that will expose the `daemonFactory.spawn` method
+
+- These method return a factory that exposes the `spawn` method, which allows spawning and controlling ipfs nodes
+
+
+> `daemonFactory.server` 
+
+- exposes `start` and `stop` methods to start and stop the bundled http server that is required to run the remote controller.
+
+#### Spawn nodes
+
+> Spawn either a js-ipfs or go-ipfs node through `localController` or `remoteController`
+
+`spawn([options], callback)`
+
+- `options` - is an optional object with various options and ipfs config parameters
+  - `js` bool (default false) - spawn a js or go node (default go)
+  - `init` bool (default true) - should the node be initialized
+  - `start` bool (default true) - should the node be started
+  - `repoPath` string - the repository path to use for this node, ignored if node is disposable
+  - `disposable` bool - a new repo is created and initialized for each invocation
+  - `args` - array of cmd line arguments to be passed to ipfs daemon
+  - `config` - ipfs configuration options
+  
+ - `callback` - is a function with the signature `cb(err, ipfsd)` where:
+   - `err` - is the error set if spawning the node is unsuccessful
+   - `ipfsd` - is an object with two properties: 
+     - `ctl` - an [ipfs-api](https://github.com/ipfs/js-ipfs-api) instance attached to the newly created ipfs node
+     - `ctrl` - an instance of a daemon controller object
+   
+### IPFS Client (ctl)
+
+> An instance of [ipfs-api](https://github.com/ipfs/js-ipfs-api#api)
```

Comment:
Explain that it is available through every daemon instance

Reply:
done